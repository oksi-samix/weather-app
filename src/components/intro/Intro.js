import React, {Component} from 'react';
import moment from 'moment';

import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';

import './intro.scss'
import WeatherList from '../WeatherList'

class Intro extends Component {

  onChange = ({target: {value, name}}) => {
    if (name === 'longitude') {
      return this.props.setLongitude(value);
    }
    if (name === 'date') {
      return this.props.setDate(value);
    }
    this.props.setLatitude(value);
  };

  getForecasted = () => {
    const date = moment(this.props.date).format('YYYY-MM-DDTHH:MM:SSZ');
    this.props.getForecasted(this.props.longitude, this.props.latitude, date)
  };

  getObserved = () => {
    const lastThirtyDays = [...new Array(30)]
      .map((i, idx) => moment(this.props.date).startOf(this.props.date).subtract(idx, "days").format('YYYY-MM-DDTHH:MM:SSZ'));
    this.props.getObserved(this.props.longitude, this.props.latitude, lastThirtyDays)
  };

  render() {
    const {getForecasted, getObserved, onChange} = this;
    const {weather, loading, error} = this.props;
    return (
      <section className="App">
        {error ? <mark className="error">Please enter longitude and longitude</mark> : null}
        {loading ? <div className="loader-box"><CircularProgress/></div> : null}
        <div>
          <TextField
            style={{padding: 24}}
            placeholder="Enter longitude"
            margin="normal"
            name="longitude"
            type="number"
            onChange={onChange}
          />
          <TextField
            style={{padding: 24}}
            placeholder="Enter latitude"
            margin="normal"
            name="latitude"
            type="number"
            onChange={onChange}
          />
          <TextField
            id="date"
            label="Pick date"
            type="date"
            defaultValue={this.props.date}
            InputLabelProps={{
              shrink: true,
            }}
            name="date"
            onChange={onChange}
          />
        </div>
        <Button onClick={getForecasted}>Forecasted daily weather</Button>
        <Button onClick={getObserved}>Observed daily weather</Button>
        {weather.length ? <WeatherList data={this.props.weather}/> : null}
      </section>
    );
  }
}

export default Intro;
