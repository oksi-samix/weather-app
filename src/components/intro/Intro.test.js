import React from 'react';
import Intro from './Intro';
import renderer from 'react-test-renderer';

it('renders Intro correctly', () => {
  const tree = renderer
    .create(<Intro weather={[]} loading={false} error={false}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
