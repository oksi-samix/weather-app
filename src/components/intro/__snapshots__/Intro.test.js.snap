// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Intro correctly 1`] = `
<section
  className="App"
>
  <div>
    <div
      className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
      style={
        Object {
          "padding": 24,
        }
      }
    >
      <div
        className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
      >
        <input
          aria-invalid={false}
          className="MuiInput-input-13 MuiInput-inputType-16"
          disabled={false}
          name="longitude"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter longitude"
          required={false}
          type="number"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
      style={
        Object {
          "padding": 24,
        }
      }
    >
      <div
        className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
      >
        <input
          aria-invalid={false}
          className="MuiInput-input-13 MuiInput-inputType-16"
          disabled={false}
          name="latitude"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter latitude"
          required={false}
          type="number"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-1"
    >
      <label
        className="MuiFormLabel-root-23 MuiInputLabel-root-18 MuiInputLabel-formControl-19 MuiInputLabel-animated-22 MuiInputLabel-shrink-21"
        data-shrink={true}
        htmlFor="date"
      >
        Pick date
      </label>
      <div
        className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
      >
        <input
          aria-invalid={false}
          className="MuiInput-input-13 MuiInput-inputType-16"
          disabled={false}
          id="date"
          name="date"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="date"
        />
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root-56 MuiButton-root-30 MuiButton-text-32 MuiButton-flat-35"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-31"
    >
      Forecasted daily weather
    </span>
    <span
      className="MuiTouchRipple-root-59"
    />
  </button>
  <button
    className="MuiButtonBase-root-56 MuiButton-root-30 MuiButton-text-32 MuiButton-flat-35"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-31"
    >
      Observed daily weather
    </span>
    <span
      className="MuiTouchRipple-root-59"
    />
  </button>
</section>
`;
